src/sandbox.rkt

```diff
@@ -139,9 +139,8 @@
   ;; TODO: Ignoring all user-provided preprocessing right now
   (define-values (alternatives preprocessing)
     (run-improve!
-     ;; If the specification is given, it is used for sampling points
-     (test-input test) (*context*) train-pcontext (*simplify-rules*)
-     (*num-iterations*) #:specification (test-spec test)))
+     (test-input test) (test-spec test) (*context*) train-pcontext
+     (*simplify-rules*) (*num-iterations*) (*node-limit*)))
```

Comment:
Likewise I think instead of passing `(*node-limit*)` through arguments, let's just read it wherever it's used. (Why not do that for `(*num-iterations*)`? Well, we should also do it for that, but I guess we don't really have to.)